crates/red_knot_python_semantic/src/semantic_index/ast_ids.rs

```diff
@@ -27,21 +28,32 @@ use crate::Db;
 #[derive(Debug, salsa::Update)]
 pub(crate) struct AstIds {
     /// Maps expressions to their expression id.
-    expressions_map: FxHashMap<ExpressionNodeKey, ScopedExpressionId>,
+    expressions: FxHashMap<ExpressionNodeKey, ScopedExpressionId>,
     /// Maps expressions which "use" a symbol (that is, [`ast::ExprName`]) to a use id.
-    uses_map: FxHashMap<ExpressionNodeKey, ScopedUseId>,
+    uses: FxHashMap<ExpressionNodeKey, ScopedUseId>,
+    /// Maps nodes that represent nested scopes to unique IDs representing those scopes.
+    eager_nested_scopes: FxHashMap<EagerNestedScopeNodeKey, ScopedEagerNestedScopeId>,
 }
 
 impl AstIds {
     fn expression_id(&self, key: impl Into<ExpressionNodeKey>) -> ScopedExpressionId {
         let key = &key.into();
-        *self.expressions_map.get(key).unwrap_or_else(|| {
+        *self.expressions.get(key).unwrap_or_else(|| {
             panic!("Could not find expression ID for {key:?}");
         })
     }
 
+    #[track_caller]
     fn use_id(&self, key: impl Into<ExpressionNodeKey>) -> ScopedUseId {
-        self.uses_map[&key.into()]
+        self.uses[&key.into()]
+    }
+
+    #[track_caller]
+    fn eager_nested_scope_id(
+        &self,
+        key: EagerNestedScopeNodeKey,
+    ) -> Option<ScopedEagerNestedScopeId> {
+        self.eager_nested_scopes.get(&key).copied()
```

Comment:
This returns an `Option` because we only register an eager nested scope in a (containing) scope if there are references that resolve there.

Reply:
This is no longer relevant now that I've changed how we store eager bindings